include/pgduckdb/catalog/pgduckdb_table.hpp

```diff
@@ -52,20 +52,4 @@ class PostgresHeapTable : public PostgresTable {
 	Oid oid;
 };
 
-class PostgresIndexTable : public PostgresTable {
-public:
-	PostgresIndexTable(Catalog &catalog, SchemaCatalogEntry &schema, CreateTableInfo &info, Cardinality cardinality,
-	                   Snapshot snapshot, Path *path, PlannerInfo *planner_info);
-
-public:
-	// -- Table API --
-	unique_ptr<BaseStatistics> GetStatistics(ClientContext &context, column_t column_id) override;
-	TableFunction GetScanFunction(ClientContext &context, unique_ptr<FunctionData> &bind_data) override;
-	TableStorageInfo GetStorageInfo(ClientContext &context) override;
-
-private:
-	Path *path;
-	PlannerInfo *planner_info;
```

Comment:
Do we still need the PlannerInfo to be passed into the sequence scan now? i.e. I think we were only using it to choose between sequence scan and index scan, so I think we can get rid of storing it in the `"postgres_state"` key of `context->registered_state`.

Reply:
I think you're right, I think that might also remove the entire need for `PostgresContextState`, I believe `m_rtables` and `m_needed_columns` as well as `m_query_string` are not used ?

Reply:
I'm also fine with doing that in a follow up PR btw.